lib/block-supports/media.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Media block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the style block attribute for block types that support it.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_media_support( $block_type ) {
+	// Setup attributes and styles within that if needed.
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	// Check for existing style attribute definition e.g. from block.json.
+	if ( array_key_exists( 'style', $block_type->attributes ) ) {
+		return;
+	}
+
+	$has_media_support = block_has_support( $block_type, array( 'media' ), false );
+
+	if ( $has_media_support ) {
+		$block_type->attributes['style'] = array(
+			'type' => 'object',
+		);
+	}
+}
+
+/**
+ * Renders the media styles to the block wrapper.
+ * This block support uses the `render_block` hook to ensure that
+ * it is also applied to non-server-rendered blocks.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_media_support( $block_content, $block ) {
+	$block_type                   = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$block_attributes             = $block['attrs'];
+	$has_background_image_support = block_has_support( $block_type, array( 'media', 'backgroundImage' ), false );
+
+	if (
+		! $has_background_image_support ||
+		wp_should_skip_block_supports_serialization( $block_type, 'media', 'backgroundImage' )
+	) {
+		return $block_content;
+	}
+
+	// TODO: should we get the style engine to handle `backgroundImage`?
```

Comment:
That would make sense to me

Reply:
Great! I made a start on looking at adding it to the style engine but ran into a couple of edge cases. I might leave things here for now and split out a separate PR for looking at adding in backgroundImage / backgroundSize to the style engine.

Reply:
Out of interest, what were the edge cases?

Reply:
> Out of interest, what were the edge cases?

It turns out the edge cases were me doing things wrong 😄

I _think_ I've sorted it out now, I've pushed a commit in https://github.com/WordPress/gutenberg/pull/53934/commits/b836572cf141b78d0a72f7c71d3a9c2dd09dddb7 that seems to work for now. Since this PR is getting pretty big, though, if the approach in this PR winds up being viable, I can split out those changes in a separate PR to make it easier for review.